controller/array_action/array_mediator_ds8k.py

```diff
@@ -546,14 +535,14 @@ def get_snapshot_by_id(self, src_snapshot_id):
         api_source_volume = self._get_api_volume_by_id(src_volume_id)
         return self._generate_snapshot_response(api_snapshot, api_source_volume.name)
 
-    def create_snapshot(self, name, volume_name, volume_context=None):
+    def create_snapshot(self, name, volume_name, pool_id=None):
         logger.info("creating snapshot '{0}' from volume '{1}'".format(name, volume_name))
-        if not volume_context:
+        if not pool_id:
             logger.error(
-                "volume_context is not specified, can not get volumes from storage."
+                "pool is not specified, can not get volumes from storage."
             )
-        pool = volume_context[config.CONTEXT_POOL]
-        target_api_volume = self._create_snapshot(name, pool, source_volume_name=volume_name)
+            raise array_errors.SnapshotNotFoundError(name)
```

Comment:
consider adding a separate Error for this scenario

Reply:
we have only PoolDoesNotExist and PoolDoesNotMatchCapabilities, and I thought that we add only cross platform  errors..

Reply:
interesting gray area.
a new exception would match the ("cross-storage") interface `pool_id` parameter, but might actually be raised only in ds8k.
IMO it's not so bad, let's number this option as number 1.

here are a few other options to consider:
2. ensure `pool_id` value (if specified) matches the volume pool on storage, in other mediators as well, to actually use the new exception across storage types*
3. raise here a mediator-specific exception (a new "layer" of exceptions, between internal-storage and cross-storage exceptions)
4. raise here a **very** general, built-in exception (e.g. `ValueError`)
5. use existing `PoolDoesNotExist` (I don't like this option)

*a note about option 2: in xiv and svc, volumes with same name cannot exist, even in different pools.
so the only scenario I can think of that it may affect, is when a user changes the pool parameter for an existing pvc.

got any other options to consider?
which option do you prefer?

Reply:
I think option 4 is my favorite.

Reply:
I actually numbered the options from best to worst in my opinion.
IMO, the main downside of option 4 is that built-in exceptions may also be raised by 3rd parties, and I assumed we would like the controller server to `except` this scenario and then return CSI `INVALID_ARGUMENT` status code.
Do you have an idea how to mitigate this downside?

Reply:
ok, done